smart-contracts/contracts/merkle-incentives/src/incentives/helpers.rs

```diff
@@ -4,329 +4,223 @@ use crate::{
     state::{CLAIMED_INCENTIVES, MERKLE_ROOT},
 };
 use cosmwasm_std::{Addr, Deps};
-use merkle::{hash::Hash, proof::Proof};
+use rs_merkle::algorithms::Sha256;
+use rs_merkle::{Hasher, MerkleProof};
 
 pub fn is_valid_claim(
     deps: Deps,
     address: Addr,
     coins: &CoinVec,
-    proof: String,
+    proof_hashes: Vec<[u8; 32]>,
+    leaf_index: usize,
+    total_leaves_count: usize,
 ) -> Result<CoinVec, ContractError> {
-    // the format of this will look like "addr1000utokena2000utokenb"
-    let claim = format!("{}{}", address.to_string(), coins.to_string());
-
     let merkle_root = MERKLE_ROOT.load(deps.storage)?;
 
-    verify_proof(&merkle_root, &proof, &claim)?;
+    // the format of this will look like "addr1000utokena2000utokenb"
+    let claim_string = format!("{}{}", address.to_string(), coins.to_string());
```

Comment:
I believe if we us `address.as_str()` here, we would get a reference to address, but still create a claim_string, meaning this function no longer needs ownership of address and we could borrow address in the function signature